.github/workflows/ci.yaml

```
@@ -1,52 +1,103 @@
+    outputs:
+      code_changed: ${{ steps.docs_check.outputs.code_changed }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if only docs changed (PR diff)
+        id: docs_check
+        shell: bash
+        run: |
```

actionlint (1.7.10)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)